src/emsc_integrator.cpp

```diff
@@ -0,0 +1,108 @@
+#include <iostream>
+#include <sstream>
+#include "emsc_integrator.h"
+
+EmscIntegrator::Exception::Exception(std::string msg, std::string v8_internal_msg)
+        : msg{msg + "\nInternal V8 error:\n" + v8_internal_msg} {}
+
+const char* EmscIntegrator::Exception::what() const noexcept {
+    return this->msg.c_str();
+}
+
+EmscIntegrator::EmscIntegrator(V8Executor& executor)
+        : executor(executor) {
```

Comment:
How about using uniform initialisation syntax?

```suggestion
        : executor{executor} {
```